src/App.tsx

```diff
@@ -1,35 +1,273 @@
+    setDirty: React.Dispatch<React.SetStateAction<boolean>>,
+    setError: React.Dispatch<React.SetStateAction<string>>
+  ) => {
+    const value = e.target.value;
+    setTeam(value);
+    const invalidChars = /[!@#$%^&*(),.?":{}|<>]/;
+    setDirty(true);
+    if (invalidChars.test(value)) {
+      setError("The name of the team must consist only of letters or numbers!");
+    } else {
+      setError("");
+    }
+  };
+
+  const ticketsHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value;
+    setTickets(value);
+    setTicketsDirty(true);
+    if (parseInt(value) < 0) {
+      setTicketsError("Quantity of tickets can not be negative!");
+    } else {
+      setTicketsError("");
+    }
+  };
+
+  const stadiumHandler = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const value = e.target.value;
+    setStadium(value);
+    setStadiumDirty(true);
+    if (!value) {
+      setStadiumError("Please, choose the stadium!");
+    } else {
+      setStadiumError("");
+    }
+  };
+
+  const blurHandler = (
+    e: React.FocusEvent<HTMLInputElement | HTMLSelectElement>
+  ) => {
+    switch (e.target.name) {
+      case "firstTeam":
+        setFirstTeamDirty(true);
+        if (!firstTeam) {
+          setFirstTeamError("Input can not be empty!");
+        }
+        break;
+      case "secondTeam":
+        setSecondTeamDirty(true);
+        if (!secondTeam) {
+          setSecondTeamError("Input can not be empty!");
+        }
+        break;
+      case "tickets":
+        setTicketsDirty(true);
+        if (!tickets) {
+          setTicketsError("Input can not be empty!");
+        }
+        break;
+      case "stadium":
+        setStadiumDirty(true);
+        if (!stadium) {
+          setStadiumError("Please, choose the stadium!");
+        }
+        break;
+    }
+  };
+
+  const addButtonHandler = () => {
+    setBlocks([
+      ...blocks,
+      { id: Date.now(), firstTeam, secondTeam, tickets, stadium },
+    ]);
+    setIsVisible(true);
+  };
+
+  const resetHandler = () => {
+    setFirstTeam("");
+    setFirstTeamError("");
+    setSecondTeam("");
+    setSecondTeamError("");
+    setTickets("");
+    setTicketsError("");
+    setStadium("");
+    setStadiumError("");
+  };
+
+  const removeBlock = (id: number) => {
+    setBlocks(blocks.filter((block) => block.id !== id));
+  };
 
   return (
     <>
-      <div>
-        <a href="https://vitejs.dev" target="_blank">
-          <img src={viteLogo} className="logo" alt="Vite logo" />
-        </a>
-        <a href="https://react.dev" target="_blank">
-          <img src={reactLogo} className="logo react" alt="React logo" />
-        </a>
-      </div>
-      <h1>Vite + React</h1>
-      <div className="card">
-        <button onClick={() => setCount((count) => count + 1)}>
-          count is {count}
-        </button>
-        <p>
-          Edit <code>src/App.tsx</code> and save to test HMR
-        </p>
+      <div className="page">
```

Comment:
As we discussed on the last call make the form as a separate component, please, as well as blocks.
Also, it would be beneficial to move textfield to a separate component and reuse it for first and second team textfields.